ml-agents/mlagents/trainers/ppo/trainer.py

```diff
@@ -264,6 +259,19 @@ def add_policy(
             )  # type: ignore
         for _reward_signal in self.optimizer.reward_signals.keys():
             self.collected_rewards[_reward_signal] = defaultdict(lambda: 0)
+
+        if self.saver is None:
```

Comment:
This works for ppo/sac, but it won't scale to a multiagent trainer which can expect multiple calls to add_policy for _different_ policies i.e. different savers. Should we think about that now or wait until we have a multiagent trainer?

Reply:
Yes this follows the original logic of only saving the first one and I was also thinking about this. But I can imagine this can be a relatively easy change: with a flag of whether the policy should be saved, if yes, register it to the saver with its unique name. That should do the work.
If we have a picture of how the multiagent will be like (like how we specify which policy to be saved) I can modify now accordingly. Otherwise I think it's fine to leave it until we have more concrete idea.